config.yaml

```diff
@@ -71,8 +71,10 @@ insteon:
       - 37.2d.20: 'deck'
 
     # Dimming devices (outlets, wall switches, lamp modules, etc).
+    # Note: on_off_ramp_supported is assumed false if not specified.
     dimmer:
       - 3a.29.84: 'lamp2'
+        on_off_ramp_supported: True
```

Comment:
Is there anyway to determine if a device supports this by knowing the device-category, sub-category, and firmware?

I think the preference would be to do as much for the user as possible.  This probably means that we would have to expose the Ramp topic for all dimmers, even those that don't support it, since the MQTT instance is initiated at the start.  But I think we could gracefully handle a Ramp command sent to a device that doesn't support it (log a message to the user, and the set on_off as otherwise desired).  It looks like you already have code to gracefully handle this situation, so we would just need to add a log message.

We would have to build a table of device capabilities, so we would likely need input from users on what devices have this feature.  With any luck there will be a clear line where dimming devices with sub-categories greater than XX will support this capability.

Reply:
> Is there anyway to determine if a device supports this by knowing the device-category, sub-category, and firmware?

I agree that it would be better to auto-detect support for these commands.  [This page](http://www.madreporite.com/insteon/ramprate.htm) suggests that "newer devices (e.g. LampLinc V4.1 and up)" support these commands, though it wasn't immediately clear to me how to go from this single statement to a reliable firmware version cutoff.

As far as I can tell from [this table](http://www.madreporite.com/insteon/Insteon_device_list.htm), it probably means the newer non-dual-band LampLincs support the commands.  All of my dimmers here are dual-band.  Maybe that means these commands were added for a while, then dropped in the dual-band LampLincs?

Do you happen to have any of these newer non-dual-band LampLinc modules?  Or have you come across any resources that would shed more light on which other devices should support these commands?

Reply:
Resolving as support for auto-detection based on device category/sub-category has been added as of commit 9c0afc46e640c809e54ca17ab476075dcb1c3704.